docs/GITLAB.md

```diff
@@ -0,0 +1,124 @@
+# GitLab
+
+If you would like to set up OpenFaaS Cloud on GitLab follow the steps below.
+
+## Intro
+
+GitLab will work with the steps done in [README.md](/README.md) file and some additional requirements:
+
+* GitLab instance
+
+* Configured System Hook
+
+* Additional secrets containing:
+
+  * Secret token which is sent along with System Hook
+
+  * Access token with `api`, `read_repository` and `sudo` rights
+
+* Custom tag or the default `openfaas-cloud` one applied to projects which are intended for the instance
+
+### Configure the base URL to your GitLab instance
+
+In `gitlab.yml` add the base URL of your GitLab instance in field `gitlab_instance` for `system-gitlab-event` function
+
+Example:
+
+```
+...
+      gitlab_instance: "https://gitlab.example.com/"
+...
+```
+
+### Configure your System Hook
+
+In your GitLab instance enter the `Admin area` then `System Hooks`.
+
+In the URL field add the URL of your entrypoint to `gitlab-event` function. Example would be:
+
+* `https://system.domain.xyz/gitlab-event` in case you have of-router running
+
+* `https://www.domain.xyz/function/system-gitlab-event` in case you have of-router turned off
+
+Set up Secret Token along with the URL.
+
+Create `gitlab-webhook-secret` secret that contains the token mentioned above on your OpenFaaS instance the following way:
+
+Kubernetes:
+
+```
+kubectl create secret generic -n openfaas-fn gitlab-webhook-secret --from-literal gitlab-webhook-secret="<your_token_here>"
+```
+
+Swarm:
+
+```
+echo -n "<your_token_here>" | docker secret create gitlab-webhook-secret -
+```
+
+
+The supported events are currently `push` and `project_update`/`project_destroy` through the System Hook so check the `Push events` event only and then `Add system hook`
+
+### Configure your Access Token
+
+This token is mandatory as it gives access to the API from which we take the tag and recognize if you have OpenFaaS Cloud installed, also provides us with a way to clone private/internal repositories and check the groups in which you participate.
+
```

Comment:
Suggest which user should do this - perhaps a service account or administrator?

Reply:
ok

Reply:
Done 👍